analysis-engines/appsensor-analysis-reference/src/main/java/org/owasp/appsensor/analysis/ReferenceEventAnalysisEngine.java

```diff
@@ -52,57 +52,76 @@
 	@Override
 	public void analyze(Event event) {
 		
-		SearchCriteria criteria = new SearchCriteria().
-				setUser(event.getUser()).
-				setDetectionPoint(event.getDetectionPoint()).
-				setDetectionSystemIds(appSensorServer.getConfiguration().getRelatedDetectionSystems(event.getDetectionSystem()));
-
-		// find all events matching this event for this user 
-		Collection<Event> existingEvents = appSensorServer.getEventStore().findEvents(criteria);
-
-		Collection<DetectionPoint> configuredDetectionPoints = appSensorServer.getConfiguration().findDetectionPoints(event.getDetectionPoint());
-
-		if (configuredDetectionPoints.size() > 0) {
-			
-			for(DetectionPoint configuredDetectionPoint : configuredDetectionPoints) {
-				
-				// filter and count events that match this detection point (filtering by threshold) 
-				// and that are after the most recent attack (filter by timestamp)
-				int eventCount = countEvents(existingEvents, event, configuredDetectionPoint);
-				
-				// if the event count is 0, reset to 1 -> we know at least 1 event has occurred (the one we're considering)
-				// this can occur sometimes when testing with dates out of the given range or due to clock drift
-				if (eventCount == 0) {
-					eventCount = 1;
-				}
-				
-				// examples for the below code
-				// 1. count is 5, t.count is 10 (5%10 = 5, No Violation)
-				// 2. count is 45, t.count is 10 (45%10 = 5, No Violation) 
-				// 3. count is 10, t.count is 10 (10%10 = 0, Violation Observed)
-				// 4. count is 30, t.count is 10 (30%10 = 0, Violation Observed)
 		
-				int thresholdCount = configuredDetectionPoint.getThreshold().getCount();
+		Collection<DetectionPoint> configuredDetectionPoints = null;
 		
-				if (eventCount % thresholdCount == 0) {
-					logger.info("Violation Observed for user <" + event.getUser().getUsername() + "> - storing attack");
+		//check if the client has custom detection points
+		if(appSensorServer.getConfiguration().findCustomDectectionClientName(event.getDetectionSystem().getDetectionSystemId())){
```

Comment:
Here i wanted to avoid having two loops instead of one hence the call. Much more efficient with a get on a map and a loop instead of two loops would you agree?